core/graph/src/main/java/uk/gov/gchq/gaffer/graph/GraphConfig.java

```diff
@@ -0,0 +1,267 @@
+                this.config.setLibrary(config.getLibrary());
+            }
+            this.config.getHooks().addAll(config.getHooks());
+            return this;
+        }
+
+        public Builder graphId(final String graphId) {
+            config.setGraphId(graphId);
+            return this;
+        }
+
+        public Builder library(final GraphLibrary library) {
+            this.config.setLibrary(library);
+            return this;
+        }
+
+        public Builder view(final View view) {
+            this.config.setView(view);
+            return this;
+        }
+
+        public Builder view(final Path view) {
+            return view(new View.Builder().json(view).build());
+        }
+
+        public Builder view(final InputStream view) {
+            return view(new View.Builder().json(view).build());
+        }
+
+        public Builder view(final URI view) {
+            try {
+                view(StreamUtil.openStream(view));
+            } catch (final IOException e) {
+                throw new SchemaException("Unable to read view from URI: " + view, e);
+            }
+            return this;
+        }
+
+        public Builder view(final byte[] jsonBytes) {
+            return view(new View.Builder().json(jsonBytes).build());
+        }
+
+        public Builder addHooks(final Path hooksPath) {
+            if (null == hooksPath || !hooksPath.toFile().exists()) {
+                throw new IllegalArgumentException("Unable to find graph hooks file: " + hooksPath);
+            }
+            final GraphHook[] hooks;
+            try {
+                hooks = JSON_SERIALISER.deserialise(FileUtils.readFileToByteArray(hooksPath.toFile()), GraphHook[].class);
+            } catch (final IOException e) {
+                throw new IllegalArgumentException("Unable to load graph hooks file: " + hooksPath, e);
+            }
+            return addHooks(hooks);
+        }
+
+        public Builder addHook(final Path hookPath) {
+            if (null == hookPath || !hookPath.toFile().exists()) {
+                throw new IllegalArgumentException("Unable to find graph hook file: " + hookPath);
+            }
+
+            final GraphHook hook;
+            try {
+                hook = JSON_SERIALISER.deserialise(FileUtils.readFileToByteArray(hookPath.toFile()), GraphHook.class);
+            } catch (final IOException e) {
+                throw new IllegalArgumentException("Unable to load graph hook file: " + hookPath, e);
+            }
+            return addHook(hook);
+        }
+
+        public Builder addHook(final GraphHook graphHook) {
+            this.config.getHooks().add(graphHook);
+            return this;
+        }
+
+        public Builder addHooks(final GraphHook... graphHooks) {
+            Collections.addAll(this.config.getHooks(), graphHooks);
+            return this;
+        }
+
+        public GraphConfig build() {
+            if (null == config.getLibrary()) {
```

Comment:
The graph will deal with that - it will attempt to get it from a provided store and if it is still null it will throw an exception.